src/cli/update.rs

```diff
@@ -0,0 +1,386 @@
+                }
+                for (_, p) in previous_pypi_packages {
+                    diff.removed.push(Package::Pypi(p));
+                }
+
+                environment_diff.insert(platform, diff);
+            }
+
+            // Find platforms that were completely removed
+            for (platform, packages) in previous
+                .as_ref()
+                .map(|e| e.packages_by_platform())
+                .into_iter()
+                .flatten()
+                .filter(|(platform, _)| !environment_diff.contains_key(platform))
+                .collect_vec()
+            {
+                let mut diff = PackagesDiff::default();
+                for package in packages {
+                    match package {
+                        Package::Conda(p) => {
+                            diff.removed.push(Package::Conda(p));
+                        }
+                        Package::Pypi(p) => {
+                            diff.removed.push(Package::Pypi(p));
+                        }
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+
+            // Remove empty diffs
+            environment_diff.retain(|_, diff| !diff.is_empty());
+
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Find environments that were completely removed
+        for (environment_name, environment) in previous
+            .environments()
+            .filter(|(name, _)| !result.environment.contains_key(*name))
+            .collect_vec()
+        {
+            let mut environment_diff = IndexMap::new();
+            for (platform, packages) in environment.packages_by_platform() {
+                let mut diff = PackagesDiff::default();
+                for package in packages {
+                    match package {
+                        Package::Conda(p) => {
+                            diff.removed.push(Package::Conda(p));
+                        }
+                        Package::Pypi(p) => {
+                            diff.removed.push(Package::Pypi(p));
+                        }
+                    }
+                }
+                environment_diff.insert(platform, diff);
+            }
+            result
+                .environment
+                .insert(environment_name.to_string(), environment_diff);
+        }
+
+        // Remove empty environments
+        result.environment.retain(|_, diff| !diff.is_empty());
+
+        result
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.environment.is_empty()
+    }
+
+    // Format the lock-file
+    pub fn print(&self) {
+        enum Change<'i> {
+            Added(&'i Package),
+            Removed(&'i Package),
+            Changed(&'i Package, &'i Package),
+        }
+
+        fn format_package_identifier(package: &Package) -> String {
+            match package {
```

Comment:
I think it would be nice to get info about the type of package that is being printed. 

I guess prefixing with `conda:` or `pypi:` if there are pypi-dependencies defined should do the trick for me.